File: app/src/main/java/com/githang/gradledoc/chapter/ChapterActivity.java
package com.githang.gradledoc.chapter;

import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.widget.Toast;

import com.githang.gradledoc.Constants;
import com.githang.gradledoc.R;
import com.githang.gradledoc.common.BaseRefreshActivity;
import com.githang.gradledoc.datasource.AbstractResponse;
import com.githang.gradledoc.datasource.HttpProxy;
import com.umeng.analytics.MobclickAgent;

public class ChapterActivity extends BaseRefreshActivity {
    private String url;

    private HttpProxy mHttpProxy;

    private Context mContext;

    private WebView mDocView;

    private AbstractResponse mChapterHandler = new ChapterHandler() {
        @Override
        public void onUISuccess(String content) {
            mDocView.loadDataWithBaseURL(Constants.BASE_URL, content, "text/html", "charset=UTF-8", null);
        }

        @Override
        public void onUIFailed(Throwable e) {
            Toast.makeText(mContext, "" + e.getMessage(), Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onUIFinish() {
            dismissProgressDialog();
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_chapter);
        mContext = this;
        mHttpProxy = HttpProxy.getInstance(this);

        Bundle bundle = getIntent().getExtras();
        url = bundle.getString(Constants.URL);
        setTitle(bundle.getString(Constants.TITLE));
        initView();

        requestContents();
    }

    private void initView() {
        mDocView = (WebView) findViewById(R.id.doc);
        WebSettings settings = mDocView.getSettings();
        settings.setLoadWithOverviewMode(true);
        settings.setSupportZoom(true);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            mDocView.getSettings().setLayoutAlgorithm(WebSettings.LayoutAlgorithm.TEXT_AUTOSIZING);
        } else {
            mDocView.getSettings().setLayoutAlgorithm(WebSettings.LayoutAlgorithm.NORMAL);
        }
    }

    private void requestContents() {
        showProgressDialog();
        HttpProxy.getInstance(this).requestUrl(this, url, mChapterHandler);
    }

    @Override
    protected void onRefresh() {
        showProgressDialog();
        mHttpProxy.forceRequestUrl(mContext, url, mChapterHandler);
    }
}

File: app/src/main/java/com/githang/gradledoc/app/chapter/ChapterActivity.kt
package com.githang.gradledoc.app.chapter

import android.os.Build
import android.os.Bundle
import android.webkit.WebSettings
import android.webkit.WebView

import com.githang.gradledoc.Constants
import com.githang.gradledoc.R
import com.githang.gradledoc.common.BaseRefreshActivity

/**
 * 文章内容
 * @author Geek_Soledad (msdx.android@qq.com)
 */
class ChapterActivity : BaseRefreshActivity<String>() {
    private var url: String? = null

    private lateinit var docView: WebView

    private val presenter = ChapterPresenter(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_chapter)
        initView()

        val bundle = intent?.extras ?: return
        url = bundle.getString(Constants.URL)
        title = bundle.getString(Constants.TITLE)

        presenter.request(this, url ?: return)
    }

    private fun initView() {
        docView = findViewById(R.id.doc)
        val settings = docView.settings
        settings.loadWithOverviewMode = true
        settings.setSupportZoom(true)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            settings.layoutAlgorithm = WebSettings.LayoutAlgorithm.TEXT_AUTOSIZING
        } else {
            settings.layoutAlgorithm = WebSettings.LayoutAlgorithm.NORMAL
        }
    }

    override fun onRefresh() {
        presenter.request(this, url ?: return, true)
    }

    override fun onHandle(data: String) {
        docView.loadDataWithBaseURL(Constants.BASE_URL, data, "text/html", "charset=UTF-8", null)
    }

    override fun onDestroy() {
        docView.stopLoading()
        docView.destroy()
        super.onDestroy()
    }
}
